[ArrowStringArray] TST: parametrize str.split tests

simonjayhawkins opened the pull request:
the test (and benchmark) changes broken off from #41085 as a precursor to #41085 and #41372 (which currently makes changes to the str.split path, although I may break that PR up also)

simonjayhawkins committed:
[ArrowStringArray] TST: paramerterise str.splt tests

simonjayhawkins committed:
fixup test_rsplit_to_dataframe_expand

jorisvandenbossche commented on asv_bench/benchmarks/strings.py:
```diff
@@ -230,17 +230,24 @@ def time_contains(self, dtype, regex):
 
-    params = [True, False]
-    param_names = ["expand"]
+    params = (["str", "string", "arrow_string"], [None, "-", "--"], [True, False])
```
I would leave out the `pat` from this benchmark, to reduce the combinatory explosion of cases. Which pattern is being used shouldn't influence the performance of expanding or not. So I would benchmark them separately.

simonjayhawkins replied:
Indeed. This is from broken off #41085 where sep determines the path taken, either pyarrow.compute.utf8_split_whitespace, pyarrow.compute.utf8_split_whitespace, pyarrow.compute.split_pattern or object fallback.

so can remove from this PR for now.

simonjayhawkins committed:
Merge remote-tracking branch 'upstream/master' into split_tests-benchmark

simonjayhawkins commented:
these are the current timings for master (will hopefully get improvements for object fallback by breaking off changes in  #41372 and also by using pyarrow kernels #41085)

```
 25.00%] ··· strings.Split.time_rsplit                                                                                                                   ok
[ 25.00%] ··· ============== ========== ==========
              --                     expand       
              -------------- ---------------------
                  dtype         True      False   
              ============== ========== ==========
                   str        82.2±0ms   33.5±0ms 
                  string      66.9±0ms   29.9±0ms 
               arrow_string   88.0±0ms   33.7±0ms 
              ============== ========== ==========

[ 50.00%] ··· strings.Split.time_split                                                                                                                    ok
[ 50.00%] ··· ============== ========== ==========
              --                     expand       
              -------------- ---------------------
                  dtype         True      False   
              ============== ========== ==========
                   str        107±0ms    56.8±0ms 
                  string      92.4±0ms   54.3±0ms 
               arrow_string   111±0ms    58.0±0ms 
              ============== ========== ==========
```

simonjayhawkins committed:
remove pat from benchmark for now

jorisvandenbossche commented:
Thanks Simon!
